bin/vmware-host-health.rb

```diff
@@ -0,0 +1,185 @@
+#! /opt/sensu/embedded/bin/ruby
+
+require 'sensu-plugin/check/cli'
+require 'rbvmomi'
+require 'json'
+
+class EsxiCheck < Sensu::Plugin::Check::CLI
+  check_name nil
+  option :host,
+         short: '-H HOST',
+         long: '--host HOST',
+         default: 'localhost',
+         description: 'ESXi host'
+
+  option :verify_ssl,
+         description: 'Turn on/off using SSL certificate (default: false)',
+         short: '-v',
+         long: '--verify_ssl',
+         boolean: true,
+         default: false
+
+  option :username,
+         short: '-u USERNAME',
+         long: '--username USERNAME',
+         default: 'root',
+         description: 'API username'
+
+  option :password,
+         short: '-p PASSWORD',
+         long: '--password PASSWORD',
+         default: 'root',
+         description: 'API password'
+
+  option :dryrun,
+         short: '-d',
+         long: '--dryrun',
+         description: 'run without sending to sensu but see what would be sent',
+         boolean: true,
+         default: false
+
+  option :handlers,
+         description: 'Comma separated list of handlers',
+         long: '--handlers <HANDLER>',
+         proc: proc { |s| s.split(',') },
+         default: []
+
+  option :help,
+         short: '-h',
+         long: '--help',
+         description: 'Show this message',
+         on: :tail,
+         boolean: true,
+         show_options: true,
+         exit: 0
+
+  def send_client_socket(data)
+    if config[:dryrun]
+      puts data.inspect
+    else
+      sock = UDPSocket.new
+      sock.send(data + "\n", 0, '127.0.0.1', 3030)
+    end
+  end
+
+  def send_ok(check_name, msg, source)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 0,
+      'output' => "#{self.class.name} OK: #{msg}",
+      'handlers' => config[:handlers]
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def send_warning(check_name, msg, source)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 1,
+      'output' => "#{self.class.name} WARNING: #{msg}",
+      'handlers' => config[:handlers],
+      'occurrences' => 2
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def send_critical(check_name, msg, source)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 2,
+      'output' => "#{self.class.name} CRITICAL: #{msg}",
+      'handlers' => config[:handlers],
+      'occurrences' => 2
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def send_unknown(check_name, msg)
+    event = {
+      'name' => check_name,
+      'source' => source,
+      'status' => 3,
+      'output' => "#{self.class.name} UNKNOWN: #{msg}",
+      'handlers' => config[:handlers]
+    }
+    send_client_socket(event.to_json)
+  end
+
+  def cache_file(data)
```

Comment:
In the context of a docker image for example you may have a RO filesystem, should we support the use case of doing this in memory rather than writing it out to a file?

Reply:
So the cache file I agree needs to be better named. Its something I never found a way to get around without traversing into a rabbit hole. Basically alerts that get reported at the data center level do not appear to again at the datacenter level record if they got cleared or not. So without having to try find that object and see did u clear or not I found it easier to just create a "state" file that will record all alerts, then on next run check to see if they are still alerting. Its not pretty but does work. I would gladly take any other suggestions but reading through the docs I couldn't find anything more elegant.